src/_pytest/debugging.py

```diff
@@ -144,7 +145,9 @@ def _init_pdb(cls, *args, **kwargs):
                     else:
                         tw.sep(">", "PDB set_trace")
 
-            class PytestPdbWrapper(cls._pdb_cls, object):
+            pdb_cls = cls._import_pdb_cls()
+
+            class PytestPdbWrapper(pdb_cls, object):
```

Comment:
I'd also like to move the PytestPdbWrapper class out of the method, likely into a method that gets it cached based on `pdb_cls` then (typically only one cache entry then).
This is good for less overhead in general, but allows for testing if the returned instance is the same as before (useful for pdbpp's global pdb handling).
This should be done before 5.0 to avoid conflicts in this area.

Reply:
Sure, sounds reasonable. I suspect this started as a small class so it was put inside the method for tersiness, but it has grown quite a lot since then.

Please do it right after this gets merged. 👍

Reply:
After https://github.com/pytest-dev/pytest/pull/5319.